controllers/challenges.js

```diff
@@ -7,12 +11,43 @@ const {StatusCodes} = require('http-status-codes');
  * @url GET /challenges 
  * @query count - number of past challenges to return
  * @response challenges - list of current and past challenges
- * @deprecated not implemented
  */
 const getChallenges = async (req, res) => {
-    //TODO delete deprecated tag after implementing the function
-    //TODO @VEXI19 calculating exp granted for completing challenge
-    res.status(StatusCodes.NOT_IMPLEMENTED).json({message: "Not implemented"});
+    let {count} = req.query;
+    const{ user:userID } = req.body;
+    const user = await User.findById(userID).populate({
+        path:'challenges',
+        mode:'Challenge',
+        match:{'status':'ongoing'},
```

Comment:
you only geting ongoing challenges, so leater when you filter for completed you always get empty array